db/src/tx.rs

```diff
@@ -735,6 +738,7 @@ impl<'conn, 'a, W> Tx<'conn, 'a, W> where W: TransactWatcher {
             tx_id: self.tx_id,
             tx_instant,
             tempids: tempids,
+            changeset: affected_attrs,
```

Comment:
It's worth asking the question: is `changeset` (more accurately named `affected_attributes`) a field that belongs on `TxReport`?

Reply:
I didn't think so originally, but I changed it after an early review comment when you suggested that they should be kept there.

Reply:
I think it's possible that _something_ lives on `TxReport` — perhaps `transacted_datoms`, if we can keep it efficiently.

I have two concerns about this and that: memory ballooning versus hacky over-specificity.

I think keeping a set of attributes is too specific (just as updating the attribute cache inside `transact` would have been too specific), and `TxReport` is part of our external API. On the other hand, keeping all of the transacted data only to throw it on the floor because nobody needs it would be wasted memory and effort.

That's why I suggest using a watcher and tracking "txids of interest" elsewhere: the watcher approach means we don't do any work unless there are caches/observers, and we don't allocate memory unless the watcher needs to.